unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Tests/AvatarCuratorShould.cs

```diff
@@ -0,0 +1,183 @@
+
+            Assert.NotNull(bodyshape);
+            Assert.AreEqual(catalog[WearableLiterals.BodyShapes.FEMALE], bodyshape);
+
+            Assert.NotNull(eyes);
+            Assert.AreEqual(catalog["eyes_id"], eyes);
+
+            Assert.NotNull(eyebrows);
+            Assert.AreEqual(catalog["eyebrows_id"], eyebrows);
+
+            Assert.NotNull(mouth);
+            Assert.AreEqual(catalog["mouth_id"], mouth);
+
+            Assert.NotNull(wearables);
+            Assert.IsTrue(wearables.Contains(catalog["ubody_id"]));
+            Assert.IsTrue(wearables.Contains(catalog["lbody_id"]));
+        });
+
+        [UnityTest]
+        public IEnumerator FallbackToADefaultSetOfWearables() => UniTask.ToCoroutine(async () =>
+        {
+            (WearableItem bodyshape,
+                WearableItem eyes,
+                WearableItem eyebrows,
+                WearableItem mouth,
+                List<WearableItem> wearables) = await curator.Curate(new AvatarSettings { bodyshapeId = WearableLiterals.BodyShapes.FEMALE }, new [] { WearableLiterals.BodyShapes.FEMALE, "WontFindThis" });
+
+            Assert.NotNull(bodyshape);
+            Assert.AreEqual(catalog[WearableLiterals.BodyShapes.FEMALE], bodyshape);
+
+            Assert.NotNull(eyes);
+            Assert.AreEqual(catalog[DefaultWearables.GetDefaultWearable(WearableLiterals.BodyShapes.FEMALE, WearableLiterals.Categories.EYES)], eyes);
+
+            Assert.NotNull(eyebrows);
+            Assert.AreEqual(catalog[DefaultWearables.GetDefaultWearable(WearableLiterals.BodyShapes.FEMALE, WearableLiterals.Categories.EYEBROWS)], eyebrows);
+
+            Assert.NotNull(mouth);
+            Assert.AreEqual(catalog[DefaultWearables.GetDefaultWearable(WearableLiterals.BodyShapes.FEMALE, WearableLiterals.Categories.MOUTH)], mouth);
+
+            Assert.NotNull(wearables);
+            Assert.IsTrue(wearables.Contains(catalog[DefaultWearables.GetDefaultWearable(WearableLiterals.BodyShapes.FEMALE, WearableLiterals.Categories.UPPER_BODY)]));
+            Assert.IsTrue(wearables.Contains(catalog[DefaultWearables.GetDefaultWearable(WearableLiterals.BodyShapes.FEMALE, WearableLiterals.Categories.LOWER_BODY)]));
+        });
+
+        [UnityTest]
+        public IEnumerator ThrowOnCurateIfCancellationTokenIsCancelled() => UniTask.ToCoroutine(async () =>
+        {
+            CancellationTokenSource cts = new CancellationTokenSource();
+            cts.Cancel();
+
+            //Assert
+            TestUtils.ThrowsAsync<OperationCanceledException>(curator.Curate(new AvatarSettings { bodyshapeId = WearableLiterals.BodyShapes.FEMALE }, null, cts.Token));
+        });
+
+        [Test]
+        public void DisposeResolver()
+        {
+            //Act
+            curator.Dispose();
+
+            //Arrange
+            resolver.Received().Dispose();
+        }
+
+        [TearDown]
+        public void TearDown() { curator.Dispose(); }
```

Comment:
Can we move the teardown with the setup? Just my tock :P

Reply:
I like the `TearDown` at the bottom of the class (same as we do with `OnDestroy` on monobehaviours) but since you and Nico are alined on this, I'm going to move it!

Reply:
You can put it on the bottom if you want, this one was on the middle